meshtastic/mesh_interface.py

```diff
@@ -1129,16 +1134,18 @@ def _handlePacketFromRadio(self, meshPacket, hack=False):
             requestId = decoded.get("requestId")
             if requestId is not None:
                 logging.debug(f"Got a response for requestId {requestId}")
-                # We ignore ACK packets, but send NAKs and data responses to the handlers
+                # We ignore ACK packets unless the callback is named `onAckNak`
+                # or the handler is set as ackPermitted, but send NAKs and
+                # other, data-containing responses to the handlers
                 routing = decoded.get("routing")
                 isAck = routing is not None and ("errorReason" not in routing or routing["errorReason"] == "NONE")
-                if not isAck:
-                    # we keep the responseHandler in dict until we get a non ack
-                    handler = self.responseHandlers.pop(requestId, None)
-                    if handler is not None:
-                        if not isAck or (isAck and handler.__name__ == "onAckNak"):
-                            logging.debug(f"Calling response handler for requestId {requestId}")
-                            handler.callback(asDict)
+                # we keep the responseHandler in dict until we actually call it
+                handler = self.responseHandlers.get(requestId, None)
+                if handler is not None:
+                    if (not isAck) or handler.callback.__name__ == "onAckNak" or handler.ackPermitted:
+                        handler = self.responseHandlers.pop(requestId, None)
+                        logging.debug(f"Calling response handler for requestId {requestId}")
+                        handler.callback(asDict)
```

Comment:
(this is the change that fixes the bug -- previously we were not checking for `onAckNak` before shortcutting out of the handler in line 1135. This instead fetches the handler and checks if it should pass acks to it, but only removes it from the handlers if it decides it should)